pkg/kubectl/cmd/apply.go

```diff
@@ -127,6 +129,7 @@ func NewCmdApply(baseName string, f cmdutil.Factory, out, errOut io.Writer) *cob
 	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
 	cmd.Flags().Bool("all", false, "Select all resources in the namespace of the specified resource types.")
 	cmd.Flags().StringArray("prune-whitelist", []string{}, "Overwrite the default whitelist with <group/version/kind> for --prune")
+	cmd.Flags().BoolVar(&options.FailedWhenUnchanged, "exit-failure-unchanged", false, "command exit code will be 2 if no any changes were applied with fail-when-unchanged=true")
```

Comment:
I guess they want this to be a string that pass in a integer default to "2".

Reply:
@mengqiy That can be implemented, but not default to "2", if we want a customize flag, I think a "trigger" flag need at least 3 status: 1.set 2.unset 3.customized value, for example: default or user set to 0 means unset, any other values not 0 means set and with customized values.

But to be honest I don't think give user too much flexibility is always a good thing, in this case allow user to define customized exit code of unchanged operation didn't gain more user experience, also it could make the experience worse, like I want to use the "trigger" flag but if I set `exit-failure-unchanged = 0` which also means the trigger is unset which is an inconsistency experience.
I think that's maybe the reason @smarterclayton thinks the latter is better (haven't confirm with him) : https://github.com/kubernetes/kubernetes/issues/52577#issue-258177971

Reply:
Yeah, that's mostly what I was working about.  

2 is probably not ok: http://tldp.org/LDP/abs/html/exitcodes.html#EXITCODESREF

I'm looking to see whether there are conventions for this in existing, commonly used Unix commands (git, diff, ansible, etc)

Reply:
Actually, thinking through this more - I may very well want to exit with status code zero if the outcome is "no-change".

Reply:
Then we could set the default to a high number but someone who wants to ignore unchanged could also use 0.

Reply:
Your flag names are inconsistent here in the description.